lib/mastodon/migration_helpers.rb

```diff
@@ -42,8 +42,12 @@
 module Mastodon
   module MigrationHelpers
     class CorruptionError < StandardError
-      def initialize(message = nil)
-        super(message.presence || 'Migration failed because of index corruption, see https://docs.joinmastodon.org/admin/troubleshooting/index-corruption/#fixing')
+      attr_reader :index_name
+
+      def initialize(index_name)
+        @index_name = index_name
+
+        super('The #{index_name} index seems to be corrupted, it contains duplicate rows')
```

Comment:
I think that having a direct link to the documentation in the error message is better for admins… and if the URL ends up changing, we can update it.

Reply:
I’d want the link to at least be printed in the rescue block, not in the exception message itself, but I’d need to extend the migrator class for that I guess

Reply:
I don't really understand that opposition :/
As for how the migrator works, it raises a `StandardError` with the detailed user-facing message on any exception in `ActiveRecord::Migration`, so doing otherwise would need changing both `ActiveRecord::Migration` (to *not* raise a `StandardError` so that the original exception can be handled differently) and `ActiveRecord::Migrator` to actually handle it differently. I do not think this is worth it.